### PR TITLE
add expToVariable() function

### PR DESCRIPTION
**Pull request by WalterBright**
Factor out common code into new function.

### CODE REVIEW

File: src/dmd/expression.d [hunk 1]
```diff
@@ -1072,7 +1072,13 @@ extern (C++) int expandAliasThisTuples(Expressions* exps, size_t starti = 0)
 }
 
 /****************************************
- * Get TemplateDeclaration enclosing FuncDeclaration.
+ * If `s` is a function template, i.e. the only member of a template
+ * and that member is a function,
+ * return that template.
```
Comment (jacob-carlborg) [inline src/dmd/expression.d]: Why is there a newline after the comma? Looks like `return that template` would fit on the previous line.

File: src/dmd/expressionsem.d [hunk 2]
```diff
@@ -5571,6 +5542,8 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                 /* Slicing a static array is like taking the address of it.
                  * Perform checks as if e[] was &e
                  */
+version (none)
```
Comment (JinShil) [inline src/dmd/expressionsem.d]: Why not just delete this code?

Comment (WalterBright) [reply src/dmd/expressionsem.d]: Because I'm testing it.

**Comment by WalterBright** [comment]
@JinShil It's ready now.

**Review by JinShil — APPROVED** [review]

**Comment by wilzbach** [comment]
Out of interest: any reason why this isn't a member function or just `toVariable`?
`exp.expToVariable` is a bit redundant and reminds me of the old C-style function naming like `atoi`.

**Comment by WalterBright** [comment]
Too many member functions create a 'kitchen sink' class which tends to be a mess.

**Pull request merged**